src/game/server/entities/light.cpp

```diff
@@ -114,11 +114,11 @@ void CLight::Snap(int SnappingClient)
 	{
 		From = m_Pos;
 	}
-	else if(pChr && m_Layer == LAYER_SWITCH && Switchers()[m_Number].m_aStatus[pChr->Team()])
+	else if(pChr && m_Layer == LAYER_SWITCH && m_Number > 0 && Switchers()[m_Number].m_aStatus[pChr->Team()])
```

Comment:
```suggestion
	else if(pChr && m_Layer == LAYER_SWITCH && (m_Number == 0 || Switchers()[m_Number].m_aStatus[pChr->Team()]))
```

And undo the change below.

Reply:
This is wrong because of `pChr` check. The laser is always present and it won't show up in spectate

Reply:
> This is wrong because of `pChr` check. The laser is always present but it won't show up in spectate

I assumed this was changing the behavior _because_ the `pChr` check was not present anymore. If this is correct then maybe it could be clarified with a comment?